src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -130,6 +132,15 @@ public HttpConnectionPoolManager(HttpConnectionSettings settings)
                     _proxyCredentials = _proxy.Credentials ?? settings._defaultProxyCredentials;
                 }
             }
+
+            _heartBeatTimer = new Timer(static state =>
```

Comment:
This needs to be constructed similar to the other timer, in a way which avoids capturing the execution context:

https://github.com/dotnet/runtime/blob/c0707135c4969be520efae02c9593d481c57cc8f/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs#L92-L121

Reply:
The code above is already doing this; perhaps this can be done within the same try/finally. It should also reuse the `WeakReference` instance.

Reply:
I have overlook the context. Fixed.